src/avdecc/mcDomainManager.cpp

```diff
@@ -0,0 +1,443 @@
+					associatedDomains.push_back(secondaryMasterIdKV.first);
+					if (secondaryMasterIdKV.second == Error::NoError)
+					{
+						auto domainIndex = result.findDomainIndexByMasterEntityId(secondaryMasterIdKV.first);
+						if (!domainIndex)
+						{
+							// domain not created yet
+							MCDomain mediaClockDomain;
+							mediaClockDomain.setMediaClockDomainMaster(secondaryMasterIdKV.first);
+							domainIndex = result.mediaClockDomains().size();
+							mediaClockDomain.setDomainIndex(domainIndex.value());
+							result.mediaClockDomains().emplace(domainIndex.value(), mediaClockDomain);
+						}
+						associatedDomains.push_back(domainIndex.value());
+					}
+				}
+			}
+			result.entityMediaClockMasterMappings().emplace(entityId, associatedDomains);
+		}
+
+		return result;
+	}
+
+private:
+	/**
+* Finds the index of the clock stream.
+* @return If the clock stream index can not be determined,  std::numeric_limits<la::avdecc::entity::model::StreamIndex>().max() is returned.
+*/
+	std::optional<la::avdecc::entity::model::StreamIndex> findClockStreamIndex(la::avdecc::controller::model::ConfigurationNode const& configNode)
+	{
+		for (auto const& streamInputKV : configNode.streamInputs)
+		{
+			auto const& streamInput = streamInputKV.second;
+			for (auto const& streamFormat : streamInput.staticModel->formats)
+			{
+				auto const streamFormatInfo = la::avdecc::entity::model::StreamFormatInfo::create(streamFormat);
+				auto streamType = streamFormatInfo->getType();
+				if (la::avdecc::entity::model::StreamFormatInfo::Type::ClockReference == streamType)
+				{
+					return streamInputKV.first;
+				}
+			}
+		}
+		return std::nullopt;
+	}
+
+	/**
+* Compares the current mc mapping state with the previous one and 
+* emits the mediaClockConnectionsUpdate to inform the views about changes.
+*/
+	void notifyChanges()
+	{
+		// potenitally changes every other entity...
+		std::vector<la::avdecc::UniqueIdentifier> changes;
+		auto currentMCDomainMapping = createMediaClockDomainModel();
+		for (auto const& entityDomainKV : currentMCDomainMapping.entityMediaClockMasterMappings())
+		{
+			la::avdecc::UniqueIdentifier const domainMasterId = entityDomainKV.first;
+			std::vector<DomainIndex> domainIds = entityDomainKV.second;
+			if (_currentMCDomainMapping.entityMediaClockMasterMappings().count(domainMasterId))
+			{
+				auto oldDomainIndexes = _currentMCDomainMapping.entityMediaClockMasterMappings().at(domainMasterId);
+				if (oldDomainIndexes.size() > 0 && domainIds.size() > 0)
+				{
+					auto oldMaster = _currentMCDomainMapping.mediaClockDomains().at(oldDomainIndexes.at(0)); // the first index is the mc master
+					auto newMaster = currentMCDomainMapping.mediaClockDomains().at(domainIds.at(0)); // the first index is the mc master
+					if (oldMaster.mediaClockDomainMaster() != newMaster.mediaClockDomainMaster())
+					{
+						changes.push_back(domainMasterId);
+					}
+				}
+				else if (oldDomainIndexes.size() != domainIds.size())
```

Comment:
This test is only called in the case:
* one of the list is 0
* both lists are 0

I don't think it's what you actually want to test!

I suggest you use 2 variables to hold the .size() value of each list (so you don't have to call it twice), and start comparing if the counts are different, before checking if both are > 0

Reply:
My thoughts on this were:
The if checks if both contain at least one item, if not we get to the else if.
The else if therefor checks if the one of the lists is empty and the other one not. Ergo we have a change in the data.

BUT, actually this will break, in the future implementation. So will have to change it anyway.

Reply:
The code makes more sense with this explanation although I'm not certain this is exactly what is needed here (but it's just a feeling, I've not check enough the whole algo, which would be nice to be documented somewhere at the begining of the file).
This should definitely be in a separate method, whose only job is to compare the 2 vectors, with comments explaining the comparison algorithm so we remember why it was done like this in a few months :) .

Reply:
Thanks for the input :)
I don't quite get the recommendation for the method though. 
I could only imagine a method of use, if it takes two MCDomainMappings as parameters and compares those. 
The vector itsself can't be compared without the mapping data, as the indices could differ and still mean the same mc master.
So another possiblity would be a method that takes the two vectors and also two parameters of type std::unordered_map<DomainIndex, MCDomain> to check what that index refers to.
Would that fulfil your notion?